dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -93,11 +108,28 @@ public CollectionRest findOne(Context context, UUID id) {
     @Override
     public Page<CollectionRest> findAll(Context context, Pageable pageable) {
         try {
-            long total = cs.countTotal(context);
-            List<Collection> collections = cs.findAll(context, pageable.getPageSize(),
-                Math.toIntExact(pageable.getOffset()));
-            return converter.toRestPage(collections, pageable, total, utils.obtainProjection());
-        } catch (SQLException e) {
+            if (authorizeService.isAdmin(context)) {
+                long total = cs.countTotal(context);
+                List<Collection> collections = cs.findAll(context, pageable.getPageSize(),
+                    Math.toIntExact(pageable.getOffset()));
+                return converter.toRestPage(collections, pageable, total, utils.obtainProjection());
+            } else {
+                List<Collection> collections = new LinkedList<Collection>();
+                // search for all the collections and let the SOLR security plugins to limit
+                // what is returned to what the user can see
+                DiscoverQuery discoverQuery = new DiscoverQuery();
+                discoverQuery.setDSpaceObjectFilter(IndexableCollection.TYPE);
+                discoverQuery.setStart(Math.toIntExact(pageable.getOffset()));
+                discoverQuery.setMaxResults(pageable.getPageSize());
+                DiscoverResult resp = searchService.search(context, discoverQuery);
```

Comment:
I can see no discovery configuration is included here. If a local institution changes the default discovery behavior to exclude comm/coll from the discovery search results, will this functionality stop working?
(the same applies for the other searches in this PR)

Reply:
no, it will work anyway because it doesn't use a discovery configuration but build from scratch a discover query. It only use the new field intorduced via the plugin (similar to how the other security filters work)